recipes/boost-ext-ut/all/conanfile.py

```diff
@@ -59,42 +67,38 @@ def validate(self):
                         self.settings.compiler.version))
 
     def config_options(self):
-        if tools.Version(self.version) <= "1.1.8":
+        if Version(self.version) <= "1.1.8":
             del self.options.disable_module
 
     def configure(self):
-        if self.settings.compiler in ["msvc", "Visual Studio"]:
-            if "disable_module" in self.options.values:
-                self.options.disable_module = True
+        if is_msvc(self) and "disable_module" in self.options.values:
+            self.options.disable_module = True
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BOOST_UT_BUILD_BENCHMARKS"] = False
-        self._cmake.definitions["BOOST_UT_BUILD_EXAMPLES"] = False
-        self._cmake.definitions["BOOST_UT_BUILD_TESTS"] = False
-        self._cmake.definitions["PROJECT_DISABLE_VERSION_SUFFIX"] = True
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BOOST_UT_BUILD_BENCHMARKS"] = False
+        tc.cache_variables["BOOST_UT_BUILD_EXAMPLES"] = False
+        tc.cache_variables["BOOST_UT_BUILD_TESTS"] = not self.conf.get("tools.build:skip_test", default=True, check_type=bool)
+        tc.cache_variables["PROJECT_DISABLE_VERSION_SUFFIX"] = True
         disable_module = self.options.get_safe("disable_module")
         if disable_module:
-            self._cmake.definitions["BOOST_UT_DISABLE_MODULE"] = disable_module
-        self._cmake.configure()
-        return self._cmake
+            tc.cache_variables["BOOST_UT_DISABLE_MODULE"] = disable_module
+        tc.generate()
     
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE*", dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE*", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_id(self):
         self.info.header_only()
```

Comment:
```suggestion
        self.info.clear()
```
https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#the-package-id-method